### PR TITLE
Fix possible deadlock on corrupted config file

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Fix possible deadlock on corrupted config file

## Related PRs or issues

#1303
[AB#75523](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/75523)

### CODE REVIEW

File: SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs [hunk 1]
```diff
@@ -58,9 +58,13 @@ internal static AppCenter Instance
...
             get
             {
+                if (_instanceField != null)
+                {
+                    return _instanceField;
+                }
                 lock (AppCenterLock)
                 {
-                    return _instanceField ?? (_instanceField = new AppCenter());
+                    return _instanceField = new AppCenter();
```
Comment (guperrot) [inline SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs:67]: Please check https://help.semmle.com/wiki/display/CSHARP/Double-checked+lock+is+not+thread-safe and add volatile and a second if check.

Comment (guperrot) [reply SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs:67]: Or maybe we can use `Lazy`.

File: SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs [hunk 2]
```diff
@@ -74,20 +78,8 @@ internal static AppCenter Instance
...
-                }
-            }
-            set
-            {
-                lock (AppCenterLock)
-                {
-                    AppCenterLog.Level = value;
-                }
-            }
+            get => AppCenterLog.Level;
```
Comment (guperrot) [inline SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs:81]: Might need volatile

File: SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs (hunk 1 above)
Comment (MatkovIvan) [reply SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs:67]: Yeah agree, I thought about second check, and missed it yesterday

**Comment by Jamminroot** [comment]
Approved the code part. Will re-review once readme is updated and tests are added

**Comment by guperrot** [comment]
Are we sure this is the cause of the issue? Did we repro without patch and not repro with?

**Comment by MatkovIvan** [comment]
@guperrot yes

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 ## Version 3.1.1 (Under development)
 
+### App Center
+
+* **[Fix]** Possible deadlock after `AppCenter.Start` fixed.
```
Comment (MatkovIvan) [inline CHANGELOG.md:7]:
```suggestion

#### WPF

* **[Fix]** Fix possible deadlock on start if configuration file was corrupted.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,12 @@
 
 ## Version 3.1.1 (Under development)
 
+### App Center
+
+#### WPF
```
Comment (guperrot) [inline CHANGELOG.md:7]: Isn't WinForms also impacted?

Comment (MatkovIvan) [reply CHANGELOG.md:7]:
```suggestion
#### WPF/WinForms
```

**Pull request merged**